cdci_data_analysis/analysis/tokenHelper.py

```diff
@@ -42,11 +44,35 @@ def get_token_user_sending_submitted_interval_email(decoded_token):
 
 
 def get_token_user_submitted_email(decoded_token):
-    # extract user threshold
     return decoded_token['mssub'] if 'mssub' in decoded_token else None
 
+def get_token_user_done_email(decoded_token):
+    return decoded_token.get('msdone', True) # TODO: make server configurable
+
+def get_token_user_fail_email(decoded_token):
+    return decoded_token.get('msfail', True) # TODO: make server configurable
 
 def get_decoded_token(token, secret_key):
     # decode the encoded token
     if token is not None:
-        return jwt.decode(token, secret_key, algorithms=['HS256'])
\ No newline at end of file
+        return jwt.decode(token, secret_key, algorithms=[default_algorithm])
+
+def update_token(token, secret_key, payload_mutation: FunctionType):
+    token_payload = jwt.decode(token, secret_key, algorithms=[default_algorithm])    
+
+    # update here
+    # tem=300000
+
+    token_payload = payload_mutation(token_payload) 
```

Comment:
Is the goal of the `payload_mutation` to just change the content of the token payload? And sometimes also to return it?

Reply:
actually, this part it's not finished. And I did also some token inspection/modification in oda_api:
https://github.com/oda-hub/oda_api/pull/56/files which starts to duplicate this. So I remove, and think later where to put it better.